autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ClusterStatusSyncHandler.java

```diff
@@ -44,42 +62,86 @@ public void onApplicationEvent(ClusterStatusSyncEvent event) {
+        LOGGER.info("Analysing CBCluster Status '{}' for Cluster '{}. Available(Determined)={}' ", stackResponse, cluster.getStackCrn(), clusterAvailable);
 
-        updateClusterState(cluster, statusResponse, clusterAvailable);
+        updateClusterState(cluster, stackResponse, clusterAvailable);
     }
 
-    private void updateClusterState(Cluster cluster, StackStatusV4Response statusResponse, boolean clusterAvailable) {
-        if (DELETE_COMPLETED.equals(statusResponse.getStatus())) {
+    private void updateClusterState(Cluster cluster, StackV4Response stackResponse, boolean clusterAvailable) {
+        if (DELETE_COMPLETED.equals(stackResponse.getStatus())) {
             beforeDeleteCleanup(cluster);
             clusterService.removeById(cluster.getId());
-            LOGGER.info("Deleted cluster '{}', CB Stack Status '{}'.", cluster.getStackCrn(), statusResponse.getStatus());
+            LOGGER.info("Deleted cluster '{}', CB Stack Status '{}'.", cluster.getStackCrn(), stackResponse.getStatus());
         } else if (clusterAvailable && !RUNNING.equals(cluster.getState())) {
             clusterService.setState(cluster.getId(), ClusterState.RUNNING);
             LOGGER.info("Updated cluster '{}' to Running, CB Stack Status '{}', CB Cluster Status '{}'.",
-                    cluster.getStackCrn(), statusResponse.getStatus(), statusResponse.getClusterStatus());
+                    cluster.getStackCrn(), stackResponse.getStatus(), stackResponse.getCluster().getStatus());
         } else if (!clusterAvailable && RUNNING.equals(cluster.getState())) {
-            clusterService.setState(cluster.getId(), ClusterState.SUSPENDED);
-            LOGGER.info("Suspended cluster '{}', CB Stack Status '{}', CB Cluster Status '{}'",
-                    cluster.getStackCrn(), statusResponse.getStatus(), statusResponse.getClusterStatus());
+            clusterService.setState(cluster.getId(), SUSPENDED);
+            LOGGER.info("Suspended cluster '{}', CB Stack Status '{}', CB Cluster Status '{}'", cluster.getStackCrn(), stackResponse.getStatus(),
+                    stackResponse.getCluster().getStatus());
         } else if (RUNNING.equals(cluster.getState()) && (cluster.getMachineUserCrn() == null || cluster.getEnvironmentCrn() == null)) {
             populateEnvironmentAndMachineUserIfNotPresent(cluster);
         }
     }
 
+    private boolean determineClusterAvailability(StackV4Response stackResponse, DependentHostGroupsV4Response dependentHostGroupsResponse,
+            Set<String> policyHostGroups) {
+        if (undefinedDependencyPresent(dependentHostGroupsResponse, policyHostGroups)) {
+            return stackResponse.getStatus().isAvailable();
+        }
+
+        boolean dependentHostsUnhealthy = policyHostGroups.stream().anyMatch(hg -> dependentHostsUnHealthy(dependentHostGroupsResponse, stackResponse, hg));
+
+        return !(stackDeletionInProgress(stackResponse) || stackModificationInProgress(stackResponse)
```

Comment:
We need to differentiate between a stop-start scale and a regular scaling operation here.
stopstart - doesn't worry about salt operations, and hence can work when some instances are unavailable.
regular scaling (non-stopstart + schedule based) - can potentially end up needing every node in a functional state. (This will typically be an artifact of whether TARGETED_UPSCALE and UNBOUND_ELIMINATION entitlements are enabled for the cluster).

For now, I would only perform this dependent check on stopstart scaling, and file a follow-up jira to handle the entitlement based approach. To some extent - this entire determination could be made on the Cloudbreak side to say - isXOperationSupported[InCurrentClusterState].

Reply:
Updated this to distinguish between regular and stop/start scaling